src/Engine/Manager/RenderManager.cpp

```diff
@@ -314,26 +356,26 @@ void RenderManager::RenderEditorEntities(World& world, bool soundSources, bool p
     }
 }
 
-Component::Animation* RenderManager::CreateAnimation() {
-    return animations.Create();
+Component::AnimationController* RenderManager::CreateAnimation() {
+    return animationControllers.Create();
 }
 
-Component::Animation* RenderManager::CreateAnimation(const Json::Value& node) {
-    Component::Animation* animation = animations.Create();
+Component::AnimationController* RenderManager::CreateAnimation(const Json::Value& node) {
+    Component::AnimationController* animationController = animationControllers.Create();
     
-    // Load values from Json node.
-    std::string name = node.get("riggedModel", "").asString();
-    /// @todo Fix animation.
-    /*for (Geometry::Model* model : Hymn().models) {
-        if (model->name == name)
-            riggedModel = model;
-    }*/
-    
-    return animation;
+    std::string skeletonName = node.get("skeleton", "").asString();
+    if (!skeletonName.empty())
+        animationController->skeleton =  Managers().resourceManager->CreateSkeleton(skeletonName);
+
+    std::string controllerName = node.get("animationController", "").asString();
+    if (!controllerName.empty())
+        animationController->controller =  Managers().resourceManager->CreateAnimationController(controllerName);
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') The function 'GetLenses' is never used. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)